src/middleware/auth.middleware.ts

```diff
@@ -0,0 +1,26 @@
+import { Injectable, NestMiddleware, UnauthorizedException } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import passport from 'passport';
+
+import { parseAuthStrategyConfig } from '~/auth/auth.utils';
+import { User } from '~/users/user.consts';
+
+@Injectable()
+export class AuthMiddleware implements NestMiddleware {
+  private readonly strategies: string[];
+
+  constructor(configService: ConfigService) {
+    const rawAuth = configService.getOrThrow<string>('AUTH_STRATEGY');
+    this.strategies = parseAuthStrategyConfig(rawAuth);
+  }
+
+  use(req: unknown, res: unknown, next: () => void): void {
+    passport.authenticate(this.strategies, { session: false }, (request, user: User) => {
+      if (user) {
+        next();
+      } else {
+        throw new UnauthorizedException('REST API user was not found');
+      }
+    })(req, res, next);
+  }
+}
```

Comment:
My understanding is that guards are recommended for authorization, where as middleware is fine for authentication.

>Middleware is a fine choice for authentication, since things like token validation and attaching properties to the request object are not strongly connected with a particular route context (and its metadata).

https://docs.nestjs.com/guards

Reply:
https://docs.nestjs.com/security/authentication#enable-authentication-globally :)

Reply:
That link depends on having this call:

```
return ctx.getContext().req.user;
```

Between line 19,20 I would need to add: 
```ts
request.user = user // or user.id
```

and then I'd be able to use that guard. I think I need middleware and guards.

Reply:
I think I was looking at the wrong header.

I will try with the guard and see how that goes.

Reply:
It took me a while to figure out, but this is now using the guard and is much less code compared to defining custom middleware.